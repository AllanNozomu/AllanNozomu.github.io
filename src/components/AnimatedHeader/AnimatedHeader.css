.header {
    font-family: 'BebasNue';
    font-size: 7em;
    letter-spacing: 1vmin;
}
  
.different-color {
    color: rgb(235, 82, 82);
}

.animated-char span span:not(.space){
    display: inline-block; 
    transition: all 0.1s linear;
}
  
.animated-char span span:hover{
    filter: brightness(75%);
    transform: scale(1.05); 
    transform-origin: center;
}
