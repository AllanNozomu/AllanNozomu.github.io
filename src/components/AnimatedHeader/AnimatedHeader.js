import React from 'react'
import './AnimatedHeader.css';

const AnimatedChar = (c, index, cls) => {
  return (
    <span key={index} className={c === ' ' ? "space" : "animated-char " + cls}>
      {c}
    </span>
  )
}

const AnimatedHeader = (props) => {
  const { header, secondColorHeader } = props;
  return (
    <h1 className="animated-header">
      {header && 
        [...header].map((c, index) => AnimatedChar(c, index))
      }
      {secondColorHeader && 
        [...secondColorHeader].map((c, index) => AnimatedChar(c, index, "different-color"))
      }
    </h1>
  )
}

export default AnimatedHeader
