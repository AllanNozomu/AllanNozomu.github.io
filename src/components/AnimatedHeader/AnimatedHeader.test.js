import React from 'react'
import { shallow } from 'enzyme';

import AnimatedHeader from './AnimatedHeader';

// header, secondColorHeader

describe('AnimatedHeader', () => {
  it('Animated Header without params', () => {
    const component = shallow(<AnimatedHeader />);

    expect(component).toMatchSnapshot();
  });
  
  it('Animated Header with header only', () => {
    const component = shallow(<AnimatedHeader header="Hello World!" />);

    expect(component).toMatchSnapshot();
  });

  it('Animated Header with secondColorHeader only', () => {
    const component = shallow(<AnimatedHeader secondColorHeader="Hello World!" />);

    expect(component).toMatchSnapshot();
  });

  it('Animated Header', () => {
    const component = shallow(<AnimatedHeader header="Hello " secondColorHeader="World!" />);

    expect(component).toMatchSnapshot();
  });
})