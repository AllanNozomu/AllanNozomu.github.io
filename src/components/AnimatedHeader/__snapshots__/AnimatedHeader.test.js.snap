// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedHeader Animated Header 1`] = `
<span>
  <h1
    className="animated-header"
  >
    <span
      className="animated-char "
      key="0"
    >
      H
    </span>
    <span
      className="animated-char "
      key="1"
    >
      e
    </span>
    <span
      className="animated-char "
      key="2"
    >
      l
    </span>
    <span
      className="animated-char "
      key="3"
    >
      l
    </span>
    <span
      className="animated-char "
      key="4"
    >
      o
    </span>
    <span
      className="space"
      key="5"
    >
       
    </span>
    <br
      className="hidden-mobile"
    />
    <span
      className="animated-char different-color"
      key="0"
    >
      W
    </span>
    <span
      className="animated-char different-color"
      key="1"
    >
      o
    </span>
    <span
      className="animated-char different-color"
      key="2"
    >
      r
    </span>
    <span
      className="animated-char different-color"
      key="3"
    >
      l
    </span>
    <span
      className="animated-char different-color"
      key="4"
    >
      d
    </span>
    <span
      className="animated-char different-color"
      key="5"
    >
      !
    </span>
  </h1>
</span>
`;

exports[`AnimatedHeader Animated Header with header only 1`] = `
<span>
  <h1
    className="animated-header"
  >
    <span
      className="animated-char "
      key="0"
    >
      H
    </span>
    <span
      className="animated-char "
      key="1"
    >
      e
    </span>
    <span
      className="animated-char "
      key="2"
    >
      l
    </span>
    <span
      className="animated-char "
      key="3"
    >
      l
    </span>
    <span
      className="animated-char "
      key="4"
    >
      o
    </span>
    <span
      className="space"
      key="5"
    >
       
    </span>
    <span
      className="animated-char "
      key="6"
    >
      W
    </span>
    <span
      className="animated-char "
      key="7"
    >
      o
    </span>
    <span
      className="animated-char "
      key="8"
    >
      r
    </span>
    <span
      className="animated-char "
      key="9"
    >
      l
    </span>
    <span
      className="animated-char "
      key="10"
    >
      d
    </span>
    <span
      className="animated-char "
      key="11"
    >
      !
    </span>
    <br
      className="hidden-mobile"
    />
  </h1>
</span>
`;

exports[`AnimatedHeader Animated Header with secondColorHeader only 1`] = `
<span>
  <h1
    className="animated-header"
  >
    <br
      className="hidden-mobile"
    />
    <span
      className="animated-char different-color"
      key="0"
    >
      H
    </span>
    <span
      className="animated-char different-color"
      key="1"
    >
      e
    </span>
    <span
      className="animated-char different-color"
      key="2"
    >
      l
    </span>
    <span
      className="animated-char different-color"
      key="3"
    >
      l
    </span>
    <span
      className="animated-char different-color"
      key="4"
    >
      o
    </span>
    <span
      className="space"
      key="5"
    >
       
    </span>
    <span
      className="animated-char different-color"
      key="6"
    >
      W
    </span>
    <span
      className="animated-char different-color"
      key="7"
    >
      o
    </span>
    <span
      className="animated-char different-color"
      key="8"
    >
      r
    </span>
    <span
      className="animated-char different-color"
      key="9"
    >
      l
    </span>
    <span
      className="animated-char different-color"
      key="10"
    >
      d
    </span>
    <span
      className="animated-char different-color"
      key="11"
    >
      !
    </span>
  </h1>
</span>
`;

exports[`AnimatedHeader Animated Header without params 1`] = `
<span>
  <h1
    className="animated-header"
  >
    <br
      className="hidden-mobile"
    />
  </h1>
</span>
`;
